src/components/ui/Suggest.jsx

```diff
@@ -22,6 +22,7 @@ const Suggest = ({
   const [isOpen, setIsOpen] = useState(false);
   const [lastQuery, setLastQuery] = useState('');
   const isMobile = useContext(DeviceContext);
+  let isLoading = false;
```

Comment:
`isLoading` should be managed as a state value of the component.
Sorry for the bad example of `currentQuery`, this is a special case where we need to keep a kind of singleton independantly from the render cycle. It could probably be done differently, but anyway…
For your need, `useState` is the way to go :)